src/plugins/dataflow-tool/components/dataflow-program.tsx

```diff
@@ -328,7 +328,10 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
         if (!this.props.readOnly && clearHistory) {
           forEach(program.nodes, (n: Node) => {
             if (n.data.recentValues) {
-              n.data.recentValues = [];
+              n.data.recentValues = {};
+              forEach(Object.keys((n as any).data.recentValues), (v:string) => {
+                (n as any).data.recentValues[v] = [];
+              });
```

Comment:
Hmm, yeah this is definitely a mistake. I must have meant to iterate over `Object.keys(n.data.watchedValues)`, but I'll look into it more closely.

I sprinkled `(n as any)` in to avoid type complaints but I'll go through and clear out unnecessary ones and see if I can solve any legitimate ones.